### PR TITLE
Add route and layout for unmatched path

### PR DESCRIPTION
**Pull request by joshuatf**
Adds a route for unmatched paths and routes where the user does not have the capability.

**Question:** This behavior mimics that of what exists in WP core, by showing the `Sorry, you are not allowed to access this page.` message regardless of whether or not permission is an issue or if the page simply does not exist.  Do we want to continue replicating this behavior or have separate messages for genuinely not found versus missing capabilities?

### Screenshots

Screen Shot 2021-08-12 at 10 51 30 AM

### Detailed test instructions:

1. Create a user that can `read`, `view_woocommerce_reports`, but cannot `manage_woocommerce`.
2. Visit `wp-admin/admin.php?page=wc-admin`
3. Note that the error message is shown
4. Visit a bad analytics report path.  E.g., `wp-admin/admin.php?page=wc-admin&path=/analytics/nope`
5. Check that the error message is shown

### CODE REVIEW

**Comment by joelclimbsthings** [comment]
Thanks @joshuatf , code looks good! I'm getting somewhat scattered behavior for the two contexts. 

Navigating to `wp-admin/admin.php?page=wc-admin` seems to work as expected 👍🏻 . 


When navigating to an incorrect report title, I'm getting a blank screen with the standard navigation:


image


Oddly enough, if I enable the woo navigation I'm getting the error as expected, but the styling is a bit off:

image

File: client/layout/NoMatch.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
```
Comment (joelclimbsthings) [inline client/layout/NoMatch.tsx:1]: Typescript 🥳

**Comment by joelclimbsthings** [comment]
>This behavior mimics that of what exists in WP core, by showing the Sorry, you are not allowed to access this page. message regardless of whether or not permission is an issue or if the page simply does not exist. Do we want to continue replicating this behavior or have separate messages for genuinely not found versus missing capabilities?

I'm not too opinionated on this. As a user (or perhaps as a developer) I prefer to be able to see the different between these two cases. That said, there are potentially reasons to keep this a bit more obscure for security and such, although based on my limited knowledge the difference here is pretty negligible.

**Comment by joshuatf** [comment]
>When navigating to an incorrect report title, I'm getting a blank screen with the standard navigation:

@joelclimbsthings What was the path you used? This can be finicky because WCA only loads on registered or "embed" pages.  So a bad path will not usually render anything, but a bad report path should since it catches all paths under `analytics/`.

I'm open to allowing anything under `page=wc-admin` and letting the client show errors which does seem more graceful than what currently happens.

**Comment by joelclimbsthings** [comment]
> What was the path you used? This can be finicky because WCA only loads on registered or "embed" pages. So a bad path will not usually render anything, but a bad report path should since it catches all paths under analytics/.

@joshuatf I'm getting the above results with the path `admin.php?page=wc-admin&path=%2Fanalytics%2Fwhoops`. To be clear, both of the screenshots are demonstrating the same path, just with woo navigation active in the second case.

**Comment by joshuatf** [comment]
@joelclimbsthings I updated this a bit to load WCA scripts on all `wc-admin` pages, even those with incorrect paths.

I think this is generally a good path forward instead of showing blank pages, but definitely could use your 👀  as well as some others from @woocommerce/mothra before moving forward with this.

This would show the header/nav/messages on any page (e.g., `wp-admin/admin.php?page=wc-admin&path=bad`) which not only avoids blank pages but also gives easier navigation back to WooCommerce pages.

**Review by louwie17 — COMMENTED** [review]
This overall tested well, although for the `wp-admin/admin.php?page=wc-admin` page for example it still showed the `Display` and `Help` tabs in the header activity panel. Maybe we want to hide the entire header when the page isn't found?

**Comment by mattsherman** [comment]
@joshuatf Any update on this? Is this still planned to be completed?

**Comment by joshuatf** [comment]
@mattsherman Yes.  Planning to address in cooldown after all task list REST API issues are resolved.

**Review by louwie17 — APPROVED** [review]
Thanks for the updates @joshuatf, everything tested well 👍 👍 This is good to merge.

**Pull request merged**